source/declarations.tex

```diff
@@ -4239,3 +4240,414 @@
 could have the same address as \tcode{buckets}
 if their respective types are all empty.
 \end{example}
+
+\rSec2[dcl.attr.contract]{Contract attributes}%
+\indextext{attribute!contracts|(}
+
+\pnum
+Contract attributes are used to specify
+preconditions, postconditions, and assertions for functions.
+
+\begin{bnf}
+\nontermdef{contract-attribute-specifier}\br
+  \terminal{[} \terminal{[} \terminal{expects} \opt{contract-level} \terminal{:} conditional-expression \terminal{]} \terminal{]}\br
+  \terminal{[} \terminal{[} \terminal{ensures} \opt{contract-level} \terminal{:} \opt{identifier} \terminal{:} conditional-expression \terminal{]} \terminal{]}\br
```

Comment:
There's a stray colon here before the *identifier*.

Reply:
Uh, right.
So, this is a "fixup" against the original paper transcription. I'd really like to move that commit as a "fixup" right after the original paper application, but how do I do that given that I'm not supposed to force-push?

Reply:
What you did is fine, I'll figure out which commit it's a fixup for when I do the merging. Or if you're going to force push anyway to rebase and pick up the new header names table, you could merge it into the original paper commit yourself.